pkg/operator/controller/machineconfig/nodes.go

```diff
@@ -90,7 +90,7 @@ func (r *MachineConfigReconciler) ensureReqNodeLabelExists(ctx context.Context)
 
 	updNodeCount := 0
 	nodeList := &corev1.NodeList{}
-	if err := r.listWorkerNodes(ctx, nodeList); err != nil {
+	if err := r.listLabeledWorkerNodes(ctx, nodeList); err != nil {
```

Comment:
listLabeledWorkerNodes and listNoObsLabeledNodes methods won't be required now right? listNodes would be enough I think.

Reply:
listLabeledWorkerNodes and listNoObsLabeledNodes methods are still needed because this PR doesn't change the logic of labeling the nodes to be observed, it just hooks into it changing which nodes should be labeled: only those from `.spec.nodeSelector`.